### PR TITLE
added home page

### PR DESCRIPTION
**Pull request by rohanmathur91**
## Description:
- Added home page
- Added categories on the home page, clicking on that will navigate to the explore page
- Added customs hooks for changing the page title and scroll to the top.
- Added search filter on video listing page

## Link:
https://deploy-preview-7--evolt-prime.netlify.app/

## Screenshot:
chrome-capture (8) (1)

### CODE REVIEW

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -40,17 +47,21 @@ export const Navbar = () => {
...
-            type="text"
-            autoComplete="false"
-            placeholder="search..."
-            className={`${styles.search__input} border w-100 py-1 pl-6 pr-2 text-base rounded-sm`}
-          />
-        </span>
+        )}
 
         <div className={`${styles.profile__icon}`}>
           <Link to="/profile">
```
Comment (akshay-rajput) [inline src/components/Navbar/Navbar.jsx]: you can add title attribute here

File: src/reducers/videoReducer.js [hunk 2]
```diff
@@ -6,6 +6,12 @@ export const videoReducer = (videoState, { type, payload }) => {
     case "INITIALIZE_CATEGORIES":
       return { ...videoState, categories: payload };
 
+    case "SET_SEARCH_QUERY":
+      return { ...videoState, searchQuery: payload };
+
+    case "SET_CURRENT_CATEGORY":
+      return { ...videoState, currentCategory: payload };
+
     default:
```
Comment (akshay-rajput) [inline src/reducers/videoReducer.js:15]: throw new error "action type not found" in default case

**Review by Aditi0505 — APPROVED** [review]
- Good breakdown of reusable components.
- Good use of css modules.
- Good practice followed for folder structure.

**Comment by kumsomi** [comment]
- UI looks good
- Folders and name looks good
- components are working fine

**Review by kumsomi — APPROVED** [review]
- UI looks good
- Folders and name looks good
- components are working fine

**Pull request merged**